components/sync/driver/brave_sync_session_durations_metrics_recorder.h

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_SYNC_DRIVER_BRAVE_SYNC_SESSION_DURATIONS_METRICS_RECODER_H_  // NOLINT
+#define BRAVE_COMPONENTS_SYNC_DRIVER_BRAVE_SYNC_SESSION_DURATIONS_METRICS_RECODER_H_  // NOLINT
+
+#include "components/sync/driver/sync_service.h"
+#include "components/sync/driver/sync_session_durations_metrics_recorder.h"
+
+namespace syncer {
+
+class BraveSyncSessionDurationsMetricsRecorder
+    : public syncer::SyncServiceObserver,
```

Comment:
Why do we need to null out `SessionDurationsMetricsRecorder`?
And if we want to replace it, we should subclass `SessionDurationsMetricsRecorder` cc @bridiver

Reply:
I agree it seems very strange that we're not subclassing `SessionDurationsMetricsRecorder`. If we did subclass https://github.com/brave/brave-core/pull/5616/files#r468189076 wouldn't be needed anyway

Reply:
and I'd also like to know why we need to do this in the first place

Reply:
@bridiver @darkdh - 

`SyncSessionDurationsMetricsRecorder` - initiates a call to `accounts.google.com/ListAccounts` on start even if the user is not logged into any google account. We could either proxy the call to `no-thanks.invalid` or null it out. This PR nulls it out because `SyncSessionDurationsMetricsRecorder` essentially logs the `signin` duration for any user which is not required. Let me know if that makes sense. 

The reason I'm not subclassing `SessionDurationsMetricsRecorder` because the request to `accounts.google.com/ListAccounts` is initiated in the constructor of `SyncSessionDurationsMetricsRecorder` - If I subclass `SyncSessionDurationsMetricsRecorder` the request will be made in any case. 

Let me know if I'm missing something.

Reply:
@jumde then I would just replace the class and not rename it

Reply:
That makes sense. I'll update. Thanks @bridiver